pkg/controller/versions/util.go

```diff
@@ -92,6 +98,12 @@ func validateGlobal(ctx context.Context, version Ver, meta *metav1.ObjectMeta, s
 	}
 	return allErrors
 }
+func checkDiscoverySelectorsisClusterWide(spec *v2.ControlPlaneSpec, Version, allErrors []error) []error {
+	if spec.MeshConfig.DiscoverySelectors != nil {
+		return append(allErrors, fmt.Errorf("the spec.meshConfig.discoverySelectors must be set for ClusterWide"))
+	}
+	return allErrors
+}
```

Comment:
No, this function is checking the wrong thing. It needs to return an error only when `mode` is `MultiTenant` and `discoverySelectors` are non-nil. If `mode` is `ClusterWide`, then `discoverySelectors` *may* be set (but don't need to be).

Reply:
So would I change ClusterWide to MultiTenant?

Reply:
I'd simply name the function `checkDiscoverySelectors`. As for the actual check: it's not really simple, as checking `spec.Mode` is not enough, because `mode` can be set in an SMCP profile/template (i.e. `mode` might be nil in the `spec`). But we probably have similar bugs elsewhere. 

@jewertow Why do we need to disallow the use of discoverySelectors in MultiTenant mode? Even if we use xns-informer, the selectors would still work, wouldn't they? Also, we allowed users to use `techPreview.meshConfig.discoverySelectors`, did we not?

If anything, we should disallow the use of `techPreview.meshConfig.discoverySelectors` and tell users to use `meshConfig.discoverySelectors` instead.

Reply:
@luksa So I changed the name of the function, but a little confused on what I should do for the `spec.mode` or is this something that will eventually be changed later since we might have it used this way other places?

Reply:
@mayleighnmyers let's wait for @jewertow to explain why we shouldn't allow discoverySelectors in multi-tenant mode. If it turns out we can allow them, we can just remove the function altogether.

Reply:
Actually, it would be great if you could try this out (set mode: MultiTenant and specify spec.meshConfig.discoverySelectors). The best way would be to add another test case to maistra-test-tool (like the one in https://github.com/maistra/maistra-test-tool/pull/519).

Reply:
I tested MultiTenant with discoverySelectors and realized we probably don't want to allow them, as Jacek suggested. So I implemented this in the validating webhook.

@jewertow Can you review this, so we can get it merged?